### ISSUE TITLE
[question] HomeSeer WD200+ report state inconsistent

### ISSUE THREAD

**Issue by i8beef** [AlCalzone/node-zwave-js#3543]
Greetings! I have most everything working now with one issue... I have three HomeSeer WD200+ dimmers that are no reporting state consistently.

They SEEM to work correctly immediately after inclusion / interview (changing states reports state change back immediately). Then as soon as I operate the switch manually, things change. Past that point, any manual manipulation sends state changes perfectly fine. But any commands sent to change state (via targetValue) no longer receive an update on currentValue..

I've had issues with HomeSeer being... selective... of how they interpret spec before, but any ideas here on work arounds?

**Comment by i8beef** [comment]
I'm gonna cross link this for background, I had similar issues with these switches originally with ZWay as well and tried asking on the [HomeSeer forums](https://forums.homeseer.com/forum/homeseer-products-services/homeseer-z-wave-products/homeseer-dimmers-switches/hs-wd200/1401355-wd200-instant-status-not-reporting), but never got a satisfactory answer. They seemed to believe this had something to do with a non-standard implementation by HomeSeer of the "Supervision" command class stuff. At the time they were able to give me a hack that let me ignore the Supervision aspect of the device inspection which let these all work fine after that and I forgot about it... I guess I kind of hoped that either HomeSeer would figure this out and correct it, or ZWay was wrong and their implementation was also not right, but this kinda cements that its probably HomeSeer doing non-standard stuff again.

Any recourse here or do I need to go bug HomeSeer again?

**Comment by i8beef** [comment]
Oh, just one other oddity: at some point some combination of healing / checking one of the nodes got it into a "failed" state in the UI.... but it was still operable, and it actually reported state reliably in this state. But I had to reboot the Pi and when it rechecked it and put it back into a "not failed" state, it went back to this.

**Comment by i8beef** [comment]
Happy to. What logs do you need, and what would you like me to do to generate them? Note there was an issue closed not long ago that I THINK might have gone into 8.5.0 that tried to deal with this HomeSeer firmware oddity, but I'm running latest and it didn't seem to fix it, even after a re interview, and a "replace" (not exclude and include).

Note: https://github.com/zwave-js/node-zwave-js/issues/3403

Thats the PR with (it looks like the 3rd or 4th) attempt to do something about these switches upstream in zwavejs... either that didn't work, or I'm doing something wrong with the reinterview / inclusion, etc. still.

Also note, when I was using these in ZWay, they actually had me hand edit the NIF after inclusion to remove something (I'd have to go look up the forum post about it) and reinterview them to get past this... Im not sure how directly you are involved with zwavejs here, so if I need to go over there and follow this through a new ticket instead let me know too.

Final note: I went digging in store files... This switch which I did run a reinterview on (I have three of these) lists in what I believe is the current .json file WITHOUT 0x6c command class (Supervision), while the other two I haven't reinterviewed DO still have the 0x6c command class... if I understand the intent of the various zwavejs PR's I'm staring at, that appears as if the incorrectly implemented command class IS being removed as expected during interview... but I'm still not getting updates from the switch.

**Comment by i8beef** [comment]
Ok, I'm attaching a driver log for two nodes, 2 and 5. Both are HomeSeer WD200+ dimmers.

[zwavejs_2021-10-17.log](https://github.com/zwave-js/zwavejs2mqtt/files/7360581/zwavejs_2021-10-17.log)

1. Node 2 has been reincluded WITHOUT security. Note this appears to be working as expected reporting state now.
2. Node 5 has been reincluded / reinterviewed and is S2 secure after the upgrade to 8.5.0 in the latest zwavejs2mqtt. It is NOT working ok still.

I did the following for Node 2: Zwave On, Zwave Off, Manual On, Manual Off, Zwave On, Zwave Off.
- Result: Everything worked as expected reporting state (insecure test)

I did the following for Node 5: Zwave On, (no report state), Manual Off, Manual On, Manual Off, Zwave On, (no report state), Manual Off (This last manual off may have been AFTER I stopped logging)
- Result: Manual events trigger state report. Zwave events do not. (secure S2 test)

On note that I think I've seen mentioned in an old zwavejs ticket: Upon reinclusion, if I operate the switch SOLELY by zwave, it reports state fine... until the first time you manually operate it, and then it will not report state anymore on zwave events.

**Comment by AlCalzone** [comment]
Ok the Supervision seems to be the problem for node 5. Can you re-interview that one and capture another log of that?

**Comment by i8beef** [comment]
So... this is interesting. I reinterviewed the node, and then I checked in the .json file to CONFIRM that the node didn't have 0x6C (Supervision)... it didn't. I then ran this test

1. Zwave on
2. Zwave off
3. Manual on
4. Manual off
5. Zwave on

Same behavior, state reports fine up until the first manual usage, and then it ONLY reports for manual interactions, not for zwave based actions (the final on)...

And then when I checked the .json file, node 5 has 0x6C supervision again..,. its like it correctly removes it during interview, but then the first interaction on the switch causes it to revert or get a new NIF from the unit or something that brings it back...

[zwavejs_2021-10-17.log](https://github.com/zwave-js/node-zwave-js/files/7360904/zwavejs_2021-10-17.log)

**Comment by AlCalzone** [comment]
Oh man whoever developed that thing deserves to be fired 🤮

The device "supports" Supervision in the sense that it will use Supervision to make sure its reports reach the controller:
```
2021-10-17T21:31:50.363Z DRIVER « [Node 005] [REQ] [BridgeApplicationCommand]
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 57
                                    └─[SupervisionCCGet]
                                      │ session id:      43
                                      │ request updates: false
                                      └─[CentralSceneCCNotification]
                                          sequence number: 11
                                          key attribute:   KeyPressed
                                          scene number:    2
```
zwave-js takes this as a sign that the device supports Supervision (since... well... it does - it just used it). But then it does not actually support any commands over it when the driver uses it.

**Comment by i8beef** [comment]
LOL, its a shame really, they are nice switches, but yeah, HomeSeer hasn't been very interested in acknowledging these do things weird... I suspect it has something to do with "admitting we made a mistake that requires a firmware update" and "charging for a tool to do firmware updates" being in conflict 😄 

I have had open issues with them, as have others, for over a year with no response unfortunately... I gather a lot of people who choose HomeSeer switches choose HomeSeer software which has a work around for this oddity.

If you can think of a way around this without firmware updates, I'm willing to test for you. Otherwise I'll just reinclude them unsecure and call if a day... I'll also let ya know here if they ever get back to me on my latest ticket I just opened this week again (since apparently my old one disappeared magically 😄 )

**Comment by AlCalzone** [comment]
Are you able to test a custom Docker image with the changes? https://zwave-js.github.io/zwavejs2mqtt/#/development/custom-docker
Use `master` for zwavejs2mqtt, `supervision-stuff` for `node-zwave-js`.

After starting the custom image, you'll need to re-interview the device again.

**Comment by i8beef** [comment]
Sure, I've just never done this... I assume I just run that Docker.contrib script with a name like 'zwavejstest123', stop my running instance, and then do I need to copy my volume so as not to mess up the current one, or do I just repoint it to the same volume when I create a new container with the test image?

**Comment by AlCalzone** [comment]
@blhoward2 @robertsLando can you help?

**Comment by blhoward2** [comment]
Instructions here: https://zwave-js.github.io/zwavejs2mqtt/#/development/custom-docker?id=building-a-container-using-dockerfilecontrib

The Dockerfile.contrib is easiest way to build off another branch. You can just change the image in your docker-compose file. No need to change the volume.

**Comment by i8beef** [comment]
Ok, took a little doing to get it to build a node docker image on a Pi's limited resources, but I got it running and it seems to work with this... I reinterviewed and tried my same tests and it seems to give me status updates with S2 security with this. Do you need me to capture more logs or anything? If you need to cleanup that PR and have me test again or anything let me know, I'm happy to do it.

Edit: And THANK YOU for taking the time to try and fix a mistake that clearly isn't yours!

**Comment by AlCalzone** [comment]
I'd be interested in seeing a log where it works - just to be sure I didn't break something else.

**Comment by i8beef** [comment]
See attached. Did the following:

1. Reinterviewed
2. Zwave on
3. Zwave off
4. Manual on
5. Manual off
6. Zwave on
7. Zwave off

[zwavejs_2021-10-20.log](https://github.com/zwave-js/node-zwave-js/files/7384019/zwavejs_2021-10-20.log)

Edit: Interestingly, just switching back and forth between builds here, I've lost my "Product" and "Product code" on all of my devices suddenly in the UI... Reinterview didn't fix. Any idea how I get those back, or should I restore a backup here?

**Comment by i8beef** [comment]
Just FYI, HomeSeer support has "escalated to the developers" along with a reminded that the HS-WD200 switches have been discontinued and I should consider upgrading to the new 300 series switches. Thats probably a light indication that they aren't interested in fixing their firmware for these unfortunately...

**Comment by AlCalzone** [comment]
Something's weird with your log there - I only see half the messages that are obviously being exchanged. But I can see that the sent messages no longer use Supervision, so that's good.

> lost my "Product" and "Product code" on all of my devices

In zwavejs2mqtt, it sometimes looks like this when the stick hangs. Got a log of the startup?

> Thats probably a light indication that they aren't interested in fixing their firmware for these unfortunately...

Sadly... Well, that's another manufacturer on my do-not-buy list 🤷🏻‍♂️

**Comment by i8beef** [comment]
I can capture another log for you, or do you want me to change anything first?

On the missing product info, I was able to get my product ids back by restoring that backup I took before we started messing with custom containers. The manufacturer files were all there in the old directory structure, so not sure why it was confused.

**Comment by i8beef** [comment]
FYI, just upgraded to the ZWaveJS2Mqtt releases last night using 8.6. Everything is included S2 secure and reporting state correctly.

THANK YOU!

**Comment by i8beef** [comment]
Also FYI, HomeSeer support got back to me. They have declined to fix their firmware, and have instead pointed out that (a) these switches are discontinued, and I should upgrade to their new 300 line, (b) this fix is evidence that it was a problem in ZWaveJS and HASS and not their problem (despite explaining how the fix works by gimping the switch's NIF, has to be implemented in every zwave stack for their specific device, and quoting the actual zwave spec at them), and (c) an active thread on [reddit](https://www.reddit.com/r/HomeSeer/comments/qgfyq9/homeseer_giveaway_win_a_177_3pack_of_our_new/hi69ss5/) at the moment from some others apparently upset about this getting "HomeSeerMark" to light a fire under them.

They then closed my ticket for comment so I couldn't continue the conversation. I think that's a pretty good indicator that HomeSeer is happy with the state of things here, and feels its ok to lean on you and others to fix their issues for them instead of taking responsibility. Buy accordingly.
